Model/Api/Tax.php

```diff
@@ -226,6 +234,14 @@ protected function getShippingDiscount($totalsInformation, $currencyCode, $shipp
             $this->quote->getShippingAddress()
         );
 
+        // Exclude discount amount of "Fixed amount discount for whole cart" sale rule from shipping discounts.
+        // Cause the Bolt cart has full discount amount of such type of sale rule applied before collecting shipping&tax.
+        if ($cartRules = $this->cartRepository->get($this->quote->getId())->getCartFixedRules()) {
```

Comment:
Why we need to get quote from DB again?

Reply:
@dumega ---^

Reply:
@dumega Could you please explain?

Reply:
@vitaliyreznikov  @danac-gs Cause only in this way, we can get the up-to-date info of rules applied to cart, that's also M2 does in core code when calculating totals https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Checkout/Model/TotalsInformationManagement.php#L48